### ISSUE TITLE
Lesson Maintenance request: Tradução da lição Using Geospatial Data to Inform Historical Research in R

### ISSUE THREAD

**Issue by anisa-hawes** [programminghistorian/jekyll#2222]
In https://github.com/programminghistorian/ph-submissions/issues/389 @joanacvp and @dianarebelorodriguez have reported some problems with the lesson *[Using Geospatial Data to Inform Historical Research in R](https://programminghistorian.org/en/lessons/geospatial-data-analysis)*. Two issues with the lesson's code were identified during the process of translation. 

I am opening this Issue to document my process of testing the lesson and identifying a possible fix.

Diana has provided the following information:

The system used: Windows
Version numbers of the relevant software: R version 4.1.0 (2021-05-18)
The exact steps taken that caused the problem, with as much detail as possible: "I had trouble in one of the very first commands, this one: `County_Aggregate_Data <- st_read("./data/County1990ussm/")`. I solved it by writing the command like this: `County_Aggregate_Data <- st_read("Archive/geospatial-data-analysis/County1990ussm")`. With the original form, R was not recognizing the files. There was another problem, at the end of the lesson, with the last plot. I was able to do it, but the graphic result was terrible, nothing like the original image." 

A screenshot showing the plot her code produced is included in the [original Issue ticket](https://github.com/programminghistorian/ph-submissions/issues/389).

**Comment by joanacvp** [comment]
thank you very much for all your help @Anisa-ProgHist

**Comment by anisa-hawes** [comment]
Hello @joanacvp and @dianarebelorodriguez. Sincere apologies for the long delay in getting back in touch with you about this. I'm afraid I am encountering the same issue, I am unable to get R to recognise the data file in the first instance. I’m uncertain if I’ve established the working directory correctly, and the text doesn't go into any specific details about this path naming. I have reached out to colleagues on our @programminghistorian/technical-team, and am hopeful that they will be able to help with this, as well as with the problem you encountered with the plot image at the end of the tutorial.

**Comment by joanacvp** [comment]
Hello. Thank you very much for your help @Anisa-ProgHist. We will wait for the technical team feedback

**Comment by rivaquiroga** [comment]
@Anisa-ProgHist, I'm happy to help with issues related to R lessons. If you want I can take a look

**Comment by anisa-hawes** [comment]
Thank you, @rivaquiroga! That would be so much appreciated!

**Comment by rivaquiroga** [comment]
**Issue 1 (folder path)**

The instructions and code are correct **BUT** they are not easy to follow because of the structure of the zip file. When you unzip it you get `Archives/geospacial-data-analysis/`... and then the files of the lesson:

```
Archives.zip
	¦--geospacial-data-analysys 
	¦ 			¦--County1990_Data/
	¦ 			¦--County1990ussm/
	¦ 			¦--Religion/
	¦ 			¦--DP_TableDescriptions.xls
	¦ 			¦--ExtendedZIP5.csv
	¦ 			¦--GeocodedAddresses.csv
	¦ 		
	¦--__MACOSX	
```
The author is expecting you to put the files inside `geospatial-data-analysys` into a folder called `data` in your current working directory. If you do exactly exactly that, the code works. 
I suggest renaming the zip file to `data` and remove the extra folder:

```
data.zip
    ¦--County1990_Data/
    ¦--County1990ussm/
    ¦--Religion/
    ¦--DP_TableDescriptions.xls
    ¦--ExtendedZIP5.csv
    ¦--GeocodedAddresses.csv
```

This way, when people unzip it, they can move that whole folder into their working directory and have the expected path.

**Issue 2 (last plot)**

I'll check this one during the afternoon. 
~~In the meantime, @joanacvp & @dianarebelorodriguez: what do you get when you click the Zoom button in the Viewer pane? Does the plot look any better? And if you try to open it in a browser (by clicking the in the same Viewer pane the button that is at the rigth of the broom)?~~

**Comment by rivaquiroga** [comment]
**Issue 2 (last plot)**


I tried to recreate the plot and got the same problem. It looks like there have been some changes in the newer versions of plotly that broke the code.

An updated version would be:

```
p <- plot_ly(County_Aggregate_Data, type = "scatter", mode = "markers") %>%
    add_trace(x = ~(AV0AA1990/10000)/CountMembers,
              y = ~BD5AA1990,
              size = ~AV0AA1990,
              color = ~People_Urban,
              text = ~paste("AVG Incom: ",BD5AA1990 ,
                            '$<br>County:', COUNTY.y,
                            '$<br>State:', STATENAM,
                            '$<br>Members:', CountMembers),
              hoverinfo = "text") %>%
    layout(title = 'Members and Income, Size=Population',
           xaxis = list(title = 'Members per 10k population'),
           yaxis = list(title = 'Income'),
           hoverlabel = list(font = list(size = 16)))

p
```
Please note that I made two changes in the code:

1. I added two arguments to avoid getting a warning (`type = "scatter", mode = "markers"`)
2. I removed the argument that set the hover text color to black because it was illegible when the hoover box was dark purple. 

I’m not completely sure if the image in the lesson was produced with the exact same code they shared. The x axis brakes don't look like what you get by default. It would be great if the image can also be updated to show readers something more similar to what they will get:

<img width="580" alt="plotly" src="https://user-images.githubusercontent.com/31421616/131742903-e4dd96ee-495e-4226-95f3-8da097d713c2.png">

**To sum up**

These are the three changes the lesson requires:

- [x] update the zip file in the `assets` folder
- [x] update the plot code in the lesson file
- [x] update the plot image in the `images` folder

@Anisa-ProgHist, let me know if you are ok with these suggestions, and if you need any help with the PR to make the updates.

**Comment by DanielAlvesLABDH** [comment]
Dear @rivaquiroga what a great help! Many,  many thanks!

**Comment by joanacvp** [comment]
@rivaquiroga thank you very much for your precious help!
@Anisa-ProgHist how should be proceed now? Should @dianarebelorodriguez make the changes on the lesson?

**Comment by anisa-hawes** [comment]
Wow! @rivaquiroga, you are brilliant! 🌟  Thank you so much! 

I may need your help to update the assets and image content in this case. I know how to adjust the plot code within the EN lesson file itself. For the PT translation, does it make sense for @dianarebelorodriguez to incorporate these changes directly? How does it usually work for assets associated with translated lessons? 

I very much appreciate your support.

Update: In a Branch titled Issue-#2222 I have updated the EN lesson's code, making the changes you've suggested for producing this lesson's final plot. If you have time to work together, we can make the further changes required changes to the assets and images folders within this branch too. Then I'll prepare a PR to push those changes to the live site.

**Comment by rivaquiroga** [comment]
Sure, I will make the changes on that branch and let you know when I'am ready.

Regarding translated lessons: if we don't translate images and assets, the translations points to the files of the original lesson. If we do translate them, we add the new files to the original lesson folder. You can see an example [here](https://github.com/programminghistorian/jekyll/tree/gh-pages/images/building-static-sites-with-jekyll-github-pages). So yes, the PT can incorporate the changes directly into their translation.

**Comment by rivaquiroga** [comment]
@Anisa-ProgHist, the zip file and the plot are now updated in the branch` Issue-#2222`. 
Please note that I merged a PR after you created that branch, so there is chance that we will need to update your branch. If you need help with that, let me know

**Comment by joanacvp** [comment]
Hello. Thank you so much for your help once again @rivaquiroga and @Anisa-ProgHist. @dianarebelorodriguez can you please incorporate the changes in the lesson? Thank you!

**Comment by dianarebelorodriguez** [comment]
Good morning! Thank you very much @rivaquiroga and @Anisa-ProgHist for your help! Sorry, I wasn't talking a lot in here, but I'm going to change our translation later on today @joanacvp :)

**Comment by joanacvp** [comment]
Thank you so much for the great work @dianarebelorodriguez

**Comment by anisa-hawes** [comment]
Thank you for your patience, @joanacvp and @dianarebelorodriguez!

You will be able to see the specific changes made within the [[Commit History for the related PR](https://github.com/programminghistorian/jekyll/commits/Issue-%232222)]. 

One detail which I almost forgot, was updating the name of the zip file where it is mentioned in the lesson (at line 73 of the EN) so you may need to make a note to update this in your translation too. 

And thanks again to super @rivaquiroga!
